sql/sql_component.cc

```diff
@@ -234,6 +250,34 @@ bool Deployed_components::load() {
   return true;
 }
 
+#ifdef XTRABACKUP
+bool Deployed_components::load_from_memory()
+{
+  std::vector<const char *> urns;
+  auto free_memory = [&urns]() {
+    for (auto element : urns)
+      if (element != nullptr) my_free(const_cast<char *>(element));
+  };
+
+  if (make_urns(urns) == false) {
```

Comment:
what if the component string is passed ""?

Reply:
it will make the make_urns return true, however, we call this from both keyring_init(online|offline) we won't reach the point to call this function if component is empty.